### PR TITLE
Fixed carousel updating when slide count changes.

### PR DESCRIPTION
**Pull request by carlospaelinck**
- [x] Fixes edge cases around changing the number of slides dynamically
- [x] Adds tests for these cases
- [x] Cleans up demo and fixes spelling error

Addresses #301 

cc @ryan-roemer @kenwheeler

### CODE REVIEW

File: demo/app.js [hunk 1]
```diff
@@ -2,30 +2,34 @@ import Carousel from '../src/index';
...
+            </div>
           )}
         >
-          <img src="http://placehold.it/1000x400&text=slide1" />
-          <img src="http://placehold.it/1000x400&text=slide2" />
-          <img src="http://placehold.it/1000x400&text=slide3" />
-          <img src="http://placehold.it/1000x400&text=slide4" />
-          <img src="http://placehold.it/1000x400&text=slide5" />
-          <img src="http://placehold.it/1000x400&text=slide6" />
+          {[...Array(this.state.length).keys()].map(key => (
```
Comment (ryan-roemer) [inline demo/app.js]: No IE support for this? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/keys

Side note -- let's add IE tests to func tests planning :)
